File: EFCoreLibrary/Database/Models/Lesson.cs
using System.Collections.Generic;
using System;

namespace EfCoreLibrary
{
    public class Lesson : IToStringForConsole
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Hours { get; set; }
        public bool Required { get; set; }
        public int TeacherId { get; set; }

        public Teacher Teacher { get; set; }
        public ICollection<Mark> Marks { get; set; }

        public string[] getInfo()
        {
            var info = new List<string>();
            info.Add($"Id: {Id}");
            info.Add($"Name: {Name}");
            info.Add($"Hours: {Hours}");
            info.Add($"Required: " + ((Required) ? "Yes" : "No"));
            info.Add($"Teacher: {Teacher.Name} {Teacher.Surname}");
            var marks = "";
            foreach (var mark in Marks)
                marks += marks + $"{mark.Value} by {mark.Student.Name} {mark.Student.Surname} at {Convert.ToString(mark.Date).Substring(0, 10)}, ";
            info.Add($"Marks: {marks.TrimEnd(',')}");
            return info.ToArray();
        }

        public string ToStringForConsole()
        {
            return string.Format("{0,4}. {1,-16} {2,3} {3,-3}", Id, Name, Hours, (Required ? "Yes" : "No"));
        }
    }
}

File: EfCoreExample/wwwRoot/lesson.js
const uriLesson = "api/lesson";
const uriTeacher = "api/teacher";

$(document).ready(function() {
  debugger;
  urlSearch = location.search;
  if (urlSearch.indexOf("?") == 0) urlSearch = urlSearch.slice(1); // удалить первый символ ? (вопросительный знак), если он есть
  var params = new URLSearchParams(urlSearch);
  var id = params.get('id');
  getTeachersList(id);
});

function getTeachersList(id) {
  $.ajax({
    url: uriTeacher,
    type: "GET",
    accepts: "application/json",
    contentType: "application/json",
    cache: false,
    success: function(data) {
      var inputTeacher = $("#inputTeacher");
      $(inputTeacher).empty();
      data.forEach(function(item, key) {
        $(inputTeacher).append($("<option></option>").text(item.name + " " + item.surname).val(item.id));
      });
      if (id) {
        getData(id);
      }
    },
    error: function(jqXHR, textStatus, errorThrown) {
      $('#modalInfoTitle').text("Что-то пошло не так...");
      $('#modalInfo').modal('show');
    }
  });
}

function getData(id) {
  $.ajax({
    url: uriLesson + "/" + id,
    type: "GET",
    accepts: "application/json",
    contentType: "application/json",
    cache: false,
    success: function(data) {
      var lesson = data;
      $("#inputID").val(lesson.id);
      $("#inputName").val(lesson.name);
      $("#inputTeacher").val(lesson.teacherId);
      $("#inputHours").val(lesson.hours);
      document.getElementById('inputRequired').checked = lesson.required;
      divMarks = $("#divMarks");
      $(divMarks).empty();
      data.marks.forEach(function(mark, key) {
        $(divMarks)
          .append(
            $("<button type=\"button\" class=\"btn btn-sm m-1\">")
            .on("click", function() {
              window.location = 'mark.html?id=' + mark.id;
            })
            .text(mark.value + " for " + mark.student.name + " " + mark.student.surname)
          );
      });
    },
    error: function(jqXHR, textStatus, errorThrown) {
      $('#modalInfoTitle').text("Что-то пошло не так...");
      $('#modalInfo').modal('show');
    }
  });
}

function saveData() {
  var lesson = {
    id        : document.getElementById('inputID').value,
    name      : document.getElementById('inputName').value,
    teacherID : document.getElementById('inputTeacher').value,
    hours     : document.getElementById('inputHours').value,
    required  : document.getElementById('inputRequired').checked,
  };
  var id = $("#inputID").val();
  if (id) {
    $.ajax({
      url: uriLesson,
      type: "PUT",
      accepts: "application/json",
      contentType: "application/json",
      cache: false,
      data: JSON.stringify(lesson),
      success: function(result) {
        $('#modalInfoTitle').text("Изменения сохранены");
        $('#modalInfo').modal('show');
        getData(id);
      },
      error: function(jqXHR, textStatus, errorThrown) {
        $('#modalInfoTitle').text("Что-то пошло не так...");
        $('#modalInfo').modal('show');
      }
    });
  } else {
    lesson.id = 0;
    $.ajax({
      url: uriLesson,
      type: "POST",
      accepts: "application/json",
      contentType: "application/json",
      cache: false,
      data: JSON.stringify(lesson),
      success: function(result) {
        $("#alertSuccess").alert();
        $('#modalInfoTitle').text("Вы сохранили информацию о новом предмете.");
        $('#modalInfo').modal('show');
        getData(result.id);
      },
      error: function(jqXHR, textStatus, errorThrown) {
        $('#modalInfoTitle').text("Что-то пошло не так...");
        $('#modalInfo').modal('show');
      }
    });
  }
}
